### ISSUE TITLE
Apps built with AIR SDK33.1.1.86 will not work on android4.1.2 devices.

### ISSUE THREAD

**Issue by YasukazuU** [airsdk/Adobe-Runtime-Support#278]
I'm trying AIR SDK 3.1.1.86.
Apps built with AIR SDK33.1.1.86 didn't work on android4.1.2 devices.
If you build with AIR SDK33.1.1.63, it works fine.
I'm building and testing an APK with no content.

Other devices, including android4.4.2, did not experience the above issue.
Is this an isolated issue?
If anyone has tested with androi4.1.2, it would be great if you could tell us the results.

Thank you.

**Comment by ajwfrost** [comment]
Hi

The earliest device we've tested on is 4.2.2 which worked (Samsung Galaxy Tab 3 10.1").

Can you check the APK contents and the AndroidManifest.xml file to see the "uses-sdk" section?
It needs to say: ```android:minSdkVersion="14"```
which you can add in your XML descriptor file if necessary (although that should be the default, if you're building for 32-bit modes - there's no 64-bit support for Android 4.x of course).

thanks

   Andrew

**Comment by YasukazuU** [comment]
hi 

Thanks for the response.
I'm building on 32bit.
We use Animate to test this by placing only stop(); at the first frame of the timeline
I tested it with the uses-sdk below.
  <android>
    <manifestAdditions><![CDATA[
		<manifest android:installLocation="auto">
		
		<uses-sdk android:minSdkVersion="16" android:targetSdkVersion="28"/>

			<uses-permission android:name="android.permission.INTERNET"/>

		</manifest>
	]]></manifestAdditions>
  </android>

The device used is SBM200SH (android4.1.2).

Thanks

**Comment by ajwfrost** [comment]
So that API level should work on that device. Does it install properly? I.e. you can see the application icon? If so can you post the logcat output from when you try to open the app?

Thanks

**Comment by YasukazuU** [comment]
The installation was successful.
You can see the icon.
When I try to launch the app by touching the icon, it fails and I get the following message
Unfortunately,"app_name" has stopped.

Posting the logcat output

Thank you.
[log.txt](https://github.com/Gamua/Adobe-Runtime-Support/files/4399255/log.txt)

**Comment by ajwfrost** [comment]
Thanks! Interesting output, I’ll pass this on and hopefully we can find out why this is happening!!

cheers

```
I/AIR VERSION( 5206): :33.1.1.86
I/DEVICE ARCH( 5206): :armeabi-v7a
W/dalvikvm( 5206): VFY: unable to resolve virtual method 140: Landroid/app/Activity;.isInMultiWindowMode ()Z
W/dalvikvm( 5206): VFY: unable to resolve virtual method 201: Landroid/app/Activity;.shouldShowRequestPermissionRationale (Ljava/lang/String;)Z
W/dalvikvm( 5206): VFY: unable to resolve virtual method 196: Landroid/app/Activity;.requestPermissions ([Ljava/lang/String;I)V
D/skia    ( 5206): new locale en_US
W/dalvikvm( 5206): No implementation found for native Lcom/adobe/air/AIRWindowSurfaceView;.nativeSurfaceCreated:()V
W/dalvikvm( 5206): threadid=1: thread exiting with uncaught exception (group=0x40ca0438)
E/AndroidRuntime( 5206): FATAL EXCEPTION: main
E/AndroidRuntime( 5206): java.lang.UnsatisfiedLinkError: Native method not found: com.adobe.air.AIRWindowSurfaceView.nativeSurfaceCreated:()V
```

**Comment by YasukazuU** [comment]
Thank you for your support.

cheers.

**Comment by jamesCourier-Clinecal** [comment]
Seeing same problem as first reported.

When using AIR SDK 33.1.1.86, the app fails during launch on an old Samsung Galaxy S2 (Model GT-I9100) running Android 4.1.2 with the following error:

E/AndroidRuntime( 9209): FATAL EXCEPTION: main
E/AndroidRuntime( 9209): java.lang.UnsatisfiedLinkError: Native method not found: com.adobe.air.AIRWindowSurfaceView.nativeSurfaceCreated:()V
E/AndroidRuntime( 9209): at com.adobe.air.AIRWindowSurfaceView.nativeSurfaceCreated(Native Method)
E/AndroidRuntime( 9209): at com.adobe.air.AIRWindowSurfaceView.surfaceCreated(AIRWindowSurfaceView.java:730)
E/AndroidRuntime( 9209): at android.view.SurfaceView.updateWindow(SurfaceView.java:609)
E/AndroidRuntime( 9209): at android.view.SurfaceView.access$000(SurfaceView.java:86)
E/AndroidRuntime( 9209): at android.view.SurfaceView$3.onPreDraw(SurfaceView.java:178)
E/AndroidRuntime( 9209): at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:703)
E/AndroidRuntime( 9209): at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1949)
E/AndroidRuntime( 9209): at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1115)
E/AndroidRuntime( 9209): at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:4526)
E/AndroidRuntime( 9209): at android.view.Choreographer$CallbackRecord.run(Choreographer.java:725)
E/AndroidRuntime( 9209): at android.view.Choreographer.doCallbacks(Choreographer.java:555)
E/AndroidRuntime( 9209): at android.view.Choreographer.doFrame(Choreographer.java:525)
E/AndroidRuntime( 9209): at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:711)
E/AndroidRuntime( 9209): at android.os.Handler.handleCallback(Handler.java:615)
E/AndroidRuntime( 9209): at android.os.Handler.dispatchMessage(Handler.java:92)
E/AndroidRuntime( 9209): at android.os.Looper.loop(Looper.java:137)
E/AndroidRuntime( 9209): at android.app.ActivityThread.main(ActivityThread.java:4921)
E/AndroidRuntime( 9209): at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime( 9209): at java.lang.reflect.Method.invoke(Method.java:511)
E/AndroidRuntime( 9209): at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1027)
E/AndroidRuntime( 9209): at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:794)
E/AndroidRuntime( 9209): at dalvik.system.NativeStart.main(Native Method)

If I drop back to using AIR SDK 33.1.1.63 (making no other changes at all, not even to the manifest) it launches and runs fine on the same device.

**Comment by ajwfrost** [comment]
Thanks for the inputs; we believe that the problem was introduced by a recent change:
 [AIR-380] Fixing crash when using AECM microphone on Android ARMv7

This fix involved rebuilding a native Android library which we now link against. That might be causing a problem then.. so I think we'll need to rebuild this again on a lower Android version..

**Comment by Klug76** [comment]
Here is the crash report from the Amazon Kindle Fire HD 8.9 device (API level 15):
```
I/ActivityManager(  198): Start proc air.com.gs.aqualife3d.android for activity air.com.gs.aqualife3d.android/.AppEntry: pid=18545 uid=10023 gids={3003}
D/skia    (18534): SkGraphics::Term() dlclose() BLTsville (CPU) 
D/dalvikvm(  198): GC_CONCURRENT freed 1240K, 42% free 22918K/39047K, paused 2ms+5ms
D/dalvikvm(18545): GC_CONCURRENT freed 204K, 2% free 14965K/15239K, paused 2ms+2ms
D/ActivityManager(  198): Trim memory level = TRIM_MEMORY_BACKGROUND is scheduled for application: com.amazon.dcp
D/dalvikvm(18545): GC_CONCURRENT freed 146K, 2% free 15333K/15559K, paused 1ms+2ms
D/dalvikvm(18545): GC_CONCURRENT freed 150K, 2% free 15699K/15943K, paused 3ms+1ms
D/dalvikvm(18545): GC_CONCURRENT freed 144K, 2% free 16079K/16327K, paused 2ms+2ms
D/dalvikvm(18545): GC_CONCURRENT freed 165K, 2% free 16430K/16711K, paused 2ms+3ms
I/ActivityThread(18545): Pub com.facebook.app.FacebookContentProvider1433507266876699: com.facebook.FacebookContentProvider
I/ActivityThread(18545): Pub air.com.gs.aqualife3d.android.provider: com.adobe.air.CameraUIProvider
I/StartupTime1(18545): :1596186956978
D/dalvikvm(18545): DexOpt: couldn't find static field Landroid/os/Build;.SUPPORTED_ABIS
W/dalvikvm(18545): VFY: unable to resolve static field 2993 (SUPPORTED_ABIS) in Landroid/os/Build;
D/dalvikvm(18545): VFY: replacing opcode 0x62 at 0x0006
I/AIR VERSION(18545): :33.1.1.190
I/DEVICE ARCH(18545): :armeabi-v7a
I/dalvikvm(18545): Could not find method android.app.Activity.isInMultiWindowMode, referenced from method com.adobe.air.AndroidActivityWrapper.setMainView
W/dalvikvm(18545): VFY: unable to resolve virtual method 289: Landroid/app/Activity;.isInMultiWindowMode ()Z
D/dalvikvm(18545): VFY: replacing opcode 0x6e at 0x0012
I/dalvikvm(18545): Could not find method android.app.Activity.shouldShowRequestPermissionRationale, referenced from method com.adobe.air.AndroidActivityWrapper.checkPermission
W/dalvikvm(18545): VFY: unable to resolve virtual method 361: Landroid/app/Activity;.shouldShowRequestPermissionRationale (Ljava/lang/String;)Z
D/dalvikvm(18545): VFY: replacing opcode 0x6e at 0x0021
I/dalvikvm(18545): Could not find method android.app.Activity.requestPermissions, referenced from method com.adobe.air.AndroidActivityWrapper.requestPermission
W/dalvikvm(18545): VFY: unable to resolve virtual method 351: Landroid/app/Activity;.requestPermissions ([Ljava/lang/String;I)V
D/dalvikvm(18545): VFY: replacing opcode 0x6e at 0x0018
D/dalvikvm(18545): Trying to load lib /data/data/air.com.gs.aqualife3d.android/lib/libc++_shared.so 0x41999778
D/dalvikvm(18545): Added shared lib /data/data/air.com.gs.aqualife3d.android/lib/libc++_shared.so 0x41999778
D/dalvikvm(18545): No JNI_OnLoad found in /data/data/air.com.gs.aqualife3d.android/lib/libc++_shared.so 0x41999778, skipping init
D/dalvikvm(18545): Trying to load lib /data/data/air.com.gs.aqualife3d.android/lib/libCore.so 0x41999778
D/dalvikvm(18545): Trying to load lib /data/data/air.com.gs.aqualife3d.android/lib/libCore.so 0x41999778
D/dalvikvm(18545): GC_CONCURRENT freed 182K, 2% free 16791K/17095K, paused 2ms+1ms
D/dalvikvm(18545): GC_CONCURRENT freed 467K, 4% free 17039K/17607K, paused 1ms+2ms
D/dalvikvm(18545): GC_CONCURRENT freed 591K, 4% free 17254K/17927K, paused 1ms+2ms
D/dalvikvm(18545): GC_FOR_ALLOC freed 307K, 4% free 17306K/17927K, paused 20ms
I/dalvikvm-heap(18545): Grow heap (frag case) to 18.139MB for 1228816-byte allocation
D/dalvikvm(18545): GC_CONCURRENT freed <1K, 4% free 18506K/19143K, paused 2ms+2ms
D/dalvikvm(18545): GC_FOR_ALLOC freed 0K, 4% free 18506K/19143K, paused 19ms
I/dalvikvm-heap(18545): Grow heap (frag case) to 20.777MB for 2764816-byte allocation
D/dalvikvm(18545): GC_FOR_ALLOC freed 0K, 4% free 21206K/21895K, paused 18ms
V/Sensors (  198): int poll__setDelay(sensors_poll_device_t*, int, int64_t)
V/Sensors (  198): int sensors_poll_context_t::setDelay(int, int64_t)
V/Sensors (  198): setDelay - sensor Accelerometer (handle 4), rate 66 ms (15.00 Hz)
V/Sensors (  198): int poll__setDelay(sensors_poll_device_t*, int, int64_t)
V/Sensors (  198): int sensors_poll_context_t::setDelay(int, int64_t)
V/Sensors (  198): setDelay - sensor Accelerometer (handle 4), rate 66 ms (15.00 Hz)
I/PhoneWindowManager(  198): repositioning status bar
W/dalvikvm(18545): No implementation found for native Lcom/adobe/air/AIRWindowSurfaceView;.nativeSurfaceCreated ()V
D/AndroidRuntime(18545): Shutting down VM
W/dalvikvm(18545): threadid=1: thread exiting with uncaught exception (group=0x40b28330)
E/AndroidRuntime(18545): FATAL EXCEPTION: main
E/AndroidRuntime(18545): java.lang.UnsatisfiedLinkError: nativeSurfaceCreated
E/AndroidRuntime(18545): 	at com.adobe.air.AIRWindowSurfaceView.nativeSurfaceCreated(Native Method)
E/AndroidRuntime(18545): 	at com.adobe.air.AIRWindowSurfaceView.surfaceCreated(AIRWindowSurfaceView.java:732)
E/AndroidRuntime(18545): 	at android.view.SurfaceView.updateWindow(SurfaceView.java:533)
E/AndroidRuntime(18545): 	at android.view.SurfaceView.access$000(SurfaceView.java:81)
E/AndroidRuntime(18545): 	at android.view.SurfaceView$3.onPreDraw(SurfaceView.java:169)
E/AndroidRuntime(18545): 	at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:590)
E/AndroidRuntime(18545): 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1630)
E/AndroidRuntime(18545): 	at android.view.ViewRootImpl.handleMessage(ViewRootImpl.java:2462)
E/AndroidRuntime(18545): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(18545): 	at android.os.Looper.loop(Looper.java:137)
E/AndroidRuntime(18545): 	at android.app.ActivityThread.main(ActivityThread.java:4533)
E/AndroidRuntime(18545): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(18545): 	at java.lang.reflect.Method.invoke(Method.java:511)
E/AndroidRuntime(18545): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
E/AndroidRuntime(18545): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
E/AndroidRuntime(18545): 	at dalvik.system.NativeStart.main(Native Method)
D/AmazonToggle(  288): setText toggle_on
D/AmazonToggle(  288): setText toggle_off
D/QuickSettings-Wifi(  288): handleSelection Begin
W/ActivityManager(  198):   Force finishing activity air.com.gs.aqualife3d.android/.AppEntry
V/ActivityStack(  198): Prepare close transition: finishing ActivityRecord{41d90e48 air.com.gs.aqualife3d.android/.AppEntry} 
```
looks like the same problem in AIR 33.1.1.190. 
Hope for fix.

**Comment by aayoub1** [comment]
Still seeing this issue in latest AIR SDK: 33.1.1.345
Any updates?

**Comment by dmitrydmsk** [comment]
AIR SDK 3.1.1.476 still have this problem for Android 4.0.3 and probably up to 4.2.2 
What is the status of this error? We using a workaround - use AIR32 for old devices, but looks like AIR32 itself causing pretty much ANRs and Crashes. And Attempt to switch to AIR33 lead to this crash and i dont know what to say to existing users who can not afford to buy a new device. Is that problem really big and will not be fixed? I understand it is an old devices, so just need to know is there a hope for fix...
Just in case kog from device:
E/AndroidRuntime(15858): FATAL EXCEPTION: main
E/AndroidRuntime(15858): java.lang.UnsatisfiedLinkError: nativeSurfaceCreated
E/AndroidRuntime(15858): 	at com.adobe.air.AIRWindowSurfaceView.nativeSurfaceCreated(Native Method)
E/AndroidRuntime(15858): 	at com.adobe.air.AIRWindowSurfaceView.surfaceCreated(AIRWindowSurfaceView.java:732)
E/AndroidRuntime(15858): 	at android.view.SurfaceView.updateWindow(SurfaceView.java:1108)
E/AndroidRuntime(15858): 	at android.view.SurfaceView.access$000(SurfaceView.java:87)
E/AndroidRuntime(15858): 	at android.view.SurfaceView$3.onPreDraw(SurfaceView.java:183)
E/AndroidRuntime(15858): 	at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:590)
E/AndroidRuntime(15858): 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1621)
E/AndroidRuntime(15858): 	at android.view.ViewRootImpl.handleMessage(ViewRootImpl.java:2446)
E/AndroidRuntime(15858): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(15858): 	at android.os.Looper.loop(Looper.java:137)
E/AndroidRuntime(15858): 	at android.app.ActivityThread.main(ActivityThread.java:4424)
E/AndroidRuntime(15858): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(15858): 	at java.lang.reflect.Method.invoke(Method.java:511)
E/AndroidRuntime(15858): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:784)
E/AndroidRuntime(15858): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:551)
E/AndroidRuntime(15858): 	at dalvik.system.NativeStart.main(Native Method)

**Comment by ajwfrost** [comment]
The problem we have here is with the compatibility between a JNI library that was built originally in 32-bit mode, where pointers are shared between Java and C++ code (and so it was necessary to modify it for 64-bit support). The library is based on Android non-NDK source code and hence has some dependencies, which are causing the final built library to then not load properly on the older Android OS versions. And getting a build environment set up for the ICS version of Android has proved to be a big problem, since everything is now deprecated and some of the dependencies are no longer available...

We have been looking at workarounds for this. One of them is to just build a version of the runtime that excludes the dependent feature and stubs out the JNI calls - so if you can email me at andrew.frost at harman.com, I can send you a version of the Runtime.apk file that you could use in order to create an ARMv7 build that should then work on the older devices, the microphone functionality may not work properly though.

We've been discussing it again though, and are wondering whether we could revert to the original Adobe-built library here which is then only working in 32-bit mode, but to add a wrapper in at the JNI level to make that then fit with the updated Java code. So .. we can have a further look...

**Comment by ajwfrost** [comment]
Should be fixed in 33.1.1.533, please check and re-open if necessary.
Thanks

**Comment by bobrokrol** [comment]
AIR 33.1.1.713
Crash
brightness_1
Samsung Galaxy J2 Core
Android 8.1 (SDK 27)

java.lang.UnsatisfiedLinkError:
at com.adobe.flashruntime.air.VideoViewAIR.nativeSetSurfaceSize (Native Method)
at com.adobe.flashruntime.air.VideoViewAIR.access$000 (VideoViewAIR.java:35)
at com.adobe.flashruntime.air.VideoViewAIR$1.surfaceChanged (VideoViewAIR.java:58)
at android.view.SurfaceView.updateSurface (SurfaceView.java:675)
at android.view.SurfaceView$2.onPreDraw (SurfaceView.java:143)
at android.view.ViewTreeObserver.dispatchOnPreDraw (ViewTreeObserver.java:977)
at android.view.ViewRootImpl.performTraversals (ViewRootImpl.java:2407)
at android.view.ViewRootImpl.doTraversal (ViewRootImpl.java:1429)
at android.view.ViewRootImpl$TraversalRunnable.run (ViewRootImpl.java:6823)
at android.view.Choreographer$CallbackRecord.run (Choreographer.java:911)
at android.view.Choreographer.doCallbacks (Choreographer.java:723)
at android.view.Choreographer.doFrame (Choreographer.java:658)
at android.view.Choreographer$FrameDisplayEventReceiver.run (Choreographer.java:897)
at android.os.Handler.handleCallback (Handler.java:790)
at android.os.Handler.dispatchMessage (Handler.java:99)
at android.os.Looper.loop (Looper.java:164)
at android.app.ActivityThread.main (ActivityThread.java:6543)
at java.lang.reflect.Method.invoke (Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:440)
at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:810)
